debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java

```diff
@@ -81,6 +82,8 @@ public SnapshotResult execute(ChangeEventSourceContext context) throws Interrupt
 
         Connection connection = null;
 
+        delaySnapshot();
```

Comment:
This seems not bound to the historized nor even the relational change event source. So I'd suggest to create a new abstract SnapshotChangeEventSource` implementation which handles this aspect and have the existing impl inherit from that. That way we'll e.g. get this "for free" when basing MongoDB on these framework classes down the road.